crates/sui-core/src/node_sync/node_state.rs

```diff
@@ -585,8 +585,11 @@ where
         };
         match result {
             Ok(_) => Ok(SyncStatus::CertExecuted),
-            Err(SuiError::ObjectNotFound { .. }) | Err(SuiError::ObjectErrors { .. }) => {
-                debug!(?digest, "cert execution failed due to missing parents");
+            e @ (Err(SuiError::ObjectNotFound { .. }) | Err(SuiError::ObjectErrors { .. })) => {
```

Comment:
Curious how did these error checks work in the first place? Typically when there is ObjectNotFound error, it's inside a vector of an ObjectError

Reply:
In process_certificate code path, I can't find a place where `ObjectNotFound` is not insides `ObjectError`. The check is already done on the entire transactions hence multiple objects. @mystenmark  correct me if i'm wrong otherwise will remove it

Reply:
I think that's right, and the ObjectNotFound error can be removed.